runtime/compiler/optimizer/FearPointAnalysis.cpp

```diff
@@ -34,10 +34,20 @@
 #include "infra/BitVector.hpp"
 #include "infra/Checklist.hpp"
 
-bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp)
+
+bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp, TR::Node *virtualGuardNode)
    {
-   static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear") == NULL);
-   return kill && !(!comp->getOption(TR_DisableVectorAPIExpansion) && comp->getMethodSymbol()->hasVectorAPI());
+   static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear")) == NULL;
+   if (kill && !comp->getOption(TR_DisableVectorAPIExpansion))
+      {
+      TR_VirtualGuard *guardInfo = comp->findVirtualGuardInfo(virtualGuardNode);
+      if (guardInfo != NULL && guardInfo->isInlineGuard())
+         {
+         char* methodSig = guardInfo->getSymbolReference()->getSymbol()->castToMethodSymbol()->getMethod()->classNameChars();
+         return (strncmp("jdk/incubator/vector", methodSig, 20) != 0 && strncmp("jdk/internal/vm/vector", methodSig, 22) != 0);
```

Comment:
I believe that the result of `classNameChars()` is not necessarily NUL-terminated, which means you need to look at `classNameLength()` here as well, i.e. if `len = (...)->classNameLength()` and `name = (...)->classNameChars()`, then `name` starts with `"jdk/incubator/vector"` iff `len >= 20` *and* `strncmp("jdk/incubator/vector", name, 20) == 0`

Reply:
Thanks a lot for catching this, updated the check to only check the class name if the length is appropriate in https://github.com/eclipse-openj9/openj9/pull/16354/commits/65dee583088499cc7e1787e743d0191096fae72e